### PR TITLE
doc: Document Findings and the finding hash

### PR DESCRIPTION
**Pull request by kgilpin**
Add doc for the findings output and the finding hash algorithm to the README.

## TODO in the hash algorithm

- [ ] Use `normalized_sql` for `sql_query`.
- [ ] Include finding `rule` and `properties` in the digest.

## Questions about the hash algorithm

- [ ] Digest `http_client_request`.
- [ ] Include fields from the function return value?
- [ ] How is `receiver` processed into the `function` digest?
- [ ] Are HTTP headers included in the digest? If so, which ones? Just the header names, or values as well? Which values are ignored?
- [ ] `path_info` vs `normalized_path_info`?
- [ ] Sort parameters by name when computing the digest content.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,6 +61,297 @@ The configuration YAML is validated against the rule schema before the scan is r
...
+
+- For each parameter: `name` and `class`.
+- For `http_client_reqeust`: `request_method`, `url`, `headers`
+- For `http_client_response`: `status_code`, `headers`
+
+[`sql_query`](https://github.com/applandinc/appmap#sql-query-call-attributes)
+
+- SQL action, column names, and table names.
+
+_Note_ The following fields are never included in the digest:
```
Comment (torchello) [inline README.md:104]:
```suggestion
_Note:_ The following fields are never included in the digest:
```

File: README.md [hunk 2]
```diff
@@ -63,6 +61,297 @@ The configuration YAML is validated against the rule schema before the scan is r
...
 documentation for each rule to see it's pattern filters and other configurable properties.
 
+## Finding hash
+
+TODO: This section is WIP, as the hash algorithm is being upgraded.
+
+To enable de-duplication of findings, a hash is calculated for each finding. The hash is the
+`sha256` digest of a canonical content string for the finding.
+
+The digest is then the `sha256` of the content string, in hex.
```
Comment (torchello) [inline README.md:71]: It's not clear what is this `content string`? I'd assume the content contains all the properties listed below, but would be good to mention this explicitly.

**Comment by dividedmind** [comment]
I still don't think it's a good idea to document the hash at all. It's one more place to update as we change it and it is an implementation detail. I think code (with possible comments) and tests should be enough of a documentation.

**Review by dividedmind — CHANGES_REQUESTED** [review]
I still don't think it's a good idea to document the hash at all. It's one more place to update as we change it and it is an implementation detail. I think code (with possible comments) and tests should be enough of a documentation. Doubly so that it lives in two different repositories.

As for the rest of the doc, looks good to me, thanks :)

**Pull request merged**